commands/start.go

```diff
@@ -172,3 +196,54 @@ func startValidator(ctx *cli.Context) (err error) {
 
 	return
 }
+
+func getWeakSubjectivityCheckpoint(checkpointURL string) (finalizedRoot string, epoch int, err error) {
+	checkpointURL = strings.TrimRight(checkpointURL, "/")
+
+	var (
+		res     *http.Response
+		resByte []byte
+
+		slotResp apitypes.CheckpointFinalizedBlockResponse
+		rootResp apitypes.CheckpointFinalizedBlockRootResponse
+	)
+
+	res, err = http.Get(fmt.Sprintf("%s/eth/v2/beacon/blocks/finalized", checkpointURL))
+	if err != nil {
+		return
+	}
+
+	resByte, err = io.ReadAll(res.Body)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(resByte, &slotResp)
+	if err != nil {
+		return
+	}
+
+	finalizedSlot := slotResp.Data.Message.Slot
+	res, err = http.Get(fmt.Sprintf("%s/eth/v1/beacon/blocks/%s/root", checkpointURL, finalizedSlot))
+	if err != nil {
+		return
+	}
+
+	resByte, err = io.ReadAll(res.Body)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(resByte, &rootResp)
+	if err != nil {
+		return
+	}
+
+	var finalizedSlotInt int
+	finalizedSlotInt, err = strconv.Atoi(finalizedSlot)
+
+	finalizedRoot = "0x" + rootResp.Data.Root
+	epoch = finalizedSlotInt / 32
```

Comment:
Please move this `32` to consts and call it like: `slotsPerEpoch`

Reply:
Done - also wrote it in the `common` package